src/snowflake/snowpark/dataframe_transform_functions.py

```diff
@@ -57,6 +69,95 @@ def _validate_formatter_argument(self, fromatter):
         if not callable(fromatter):
             raise TypeError("formatter must be a callable function")
 
+    def _validate_and_extract_time_unit(
+        self, time_str: str, argument_name: str, allow_negative: bool = True
+    ) -> Tuple[int, str]:
+        if not time_str:
+            raise ValueError(f"{argument_name} must not be empty")
+
+        duration = int(time_str[:-1])
+        unit = time_str[-1].lower()
+
+        if not allow_negative and duration < 0:
+            raise ValueError(f"{argument_name} must not be negative.")
+
+        supported_units = ["s", "m", "h", "d", "w", "t", "y"]
+        if unit not in supported_units:
+            raise ValueError(
+                f"Unsupported unit '{unit}'. Supported units are '{supported_units}"
+            )
+
+        # Converting month unit to 'mm' for Snowpark
+        if unit == "t":
+            unit = "mm"
+
+        return duration, unit
+
+    def _get_sliding_interval_start(
+        self, time_col: Column, unit: str, duration: int
+    ) -> Column:
+        unit_seconds = {
+            "s": 1,  # seconds
+            "m": 60,  # minutes
+            "h": 3600,  # hours
+            "d": 86400,  # days
+            "w": 604800,  # weeks
+        }
+
+        if unit == "mm":
+            base_date = lit("1970-01-01").cast("date")
```

Comment:
Do we need to worry about timezone here? Also, this seems like a good static constant.